### ISSUE TITLE
My Nebraska dashboard gives me accurate estimates if I have a Center license type

### ISSUE THREAD

**Issue by csprayregen** [pieforproviders/pieforproviders#1157]
## ✨ What new feature do you want to see?
We are likely to begin working with centers as well as family child care providers in Nebraska. We should add additional complexity to our formulas to account for different license types. This is a bigger project and we should discuss specific steps within it. 

## 🖍️ Implementation Details

> **NOTE TO ENGINEERS**
> 
> Please ensure that you are running Google Lighthouse and/or WebAIM (Wave) on any UI elements in your pull request to check for accessibility issues.
> 
> If it is out of scope to address those accessibility issues in your pull request (because the issue is in a parent component and needs refactoring to work, for example), **you are responsible for noting those issues in your pull request and writing bug tickets to address the issues - make sure you link the new tickets in your pull request**

**Comment by JonErvin** [comment]
Great chat today team!  Just to recap, we'll read through the following pages and share any notes about what differs for child care centers in this thread.  Our best guess is that the rates are the largest difference, but we're looking for anything else that might affect this model.

- @joseluisrangel-dataheim Page 4-14 in the old, longer guidebook: [CCPB24](https://drive.google.com/file/d/1m_LLFU6TfGfPibUthx11NAb5IUhnrWNc/view?usp=sharing)
- @JTBassett Skim pages 15-25 and read through the Appendix forms in the old, longer guidebook: [CCPB24](https://drive.google.com/file/d/1m_LLFU6TfGfPibUthx11NAb5IUhnrWNc/view?usp=sharing)
- I will review the shorter new guidance book: [Title 392 Guidance Document](https://drive.google.com/drive/u/1/folders/1Ha_dZuKMyGBdSJH33-Y4XVVhfwF1BU0c)

Let me know if I missed anything  - we can come together on Wednesday in the standing check-in to discuss next steps!

**Comment by JTBassett** [comment]
@JonErvin I would echo your thoughts that the rates are the only data changes we need to consider for purposes of implementation. The only other difference I found is on page 14 which highlights that the care centers use a different attendance calendar (pg.39-40 as opposed to pg.37-38), with the primary differentiators being that a parent signature is not required and some of the warning messages are removed.

**Comment by JTBassett** [comment]
@JonErvin I updated the provider-level calculation tab to account for child care centers. For now, I do this by checking if the Business License Type equals "Child Care Center" and referencing Table 3 as opposed to Table 6 only in that case. Do you think I should make the check more general (I'm not sure what license names we would expect to see for these centers) or any other thoughts before I consider populating other downstream data like a schedule upload?

**Comment by JonErvin** [comment]
Hey good question... from what I know I think our users are likely to be either Family Child Care Home (I or II) or Child Care Center.  The only other relevant license type is License-Exempt Family Child Care - how difficult do you think it would be add this option as well?  If easy, let's add it as well.  If it adds a lot of complexity, I think we can hold off (not sure if we'll have any relevant users before we switch to the Ruby-powered version).

**Comment by JonErvin** [comment]
Also @JTBassett just wanted to flag that you might see a few extra fields in Provider Level Calculation related to something I'm working on for #1155 (adding date ranges to the QRIS bump / accreditation).  As far as I can tell it shouldn't impact this work, but let me know if I mess up something you were working on!

**Comment by JTBassett** [comment]
The additional fields shouldn't cause any issues! Also, I fixed some of the references that were breaking by the way.

It wouldn't be overly difficult to add the License-Exempt Family Child Care Center - another if statement would have to be added so the calculation would be pretty long, but not bad otherwise. I can still do that if you'd like. Also, do you think I should consider adding any sample data further down the line, or do you think what I've added so far (plus the License-Exempt case) would suffice for this card?

**Comment by JonErvin** [comment]
My guess is that should suffice!  If we need to test other features we can always move an existing child & schedule to one of the newly created providers in the new license types - probably a lot easier than making up new WS and schedule data

**Comment by JTBassett** [comment]
Looking at it again, I realized that the License-Exempt case breaks out the LDDS (Lancaster / Dakota v. Douglas / Sarpy) which is some complexity you may have been hinting at originally and that I missed when I actually went to implement it. I'll move this card to QA since it sounds like we won't have to worry about that case anytime in the near future, so let me know if there's any issues you find as you QA it

**Comment by JonErvin** [comment]
Hey @JTBassett will QA this is more detail in the near future, but just wanted to flag one change I made in the course of working on #1155.  To future-proof against a scenario where a provider changes license types (not sure if this is a realistic use case), I updated the formula in Provider-Level Calculation, Business License column to INDEX/MATCH on the new ProviderEntryID field.  Functionality is exactly the same but its slightly more robust - it also protects against us adding a column to Provider-Level Data (in which case the 5th column in that table might change).  Finally, I wrapped the formula in `IF(A2="", "",  ___ )` to drop it down to the last row to accommodate future provider entries.

Tl;dr - no change in functionality, but just updated this one column to be slightly more robust.  Let me know if you think could mess up anything you set up here.

**Comment by JTBassett** [comment]
Hey @JonErvin, yes I just got back from Tennessee this afternoon, but I should have time to work on the summary this weekend. I haven't updated the docs yet, but I'll do that this weekend as well.

**Comment by JTBassett** [comment]
@JonErvin I updated the docs and the summary is below for reference!

**Summary of Updates**

**Rates Table**
Added Table 9 as master Child Home Care Center Rate table for provider calculations

**Provider-Level Calculation**
Updated Infant_Hour to SchoolAge_Day columns to incorporate Child Care Center rates

**Other**
Added a fake provider and child in the Provider-Level Data and Child-Level data tabs respectively to test the updated formulas with sample data

Please let me know if you have any follow-up questions/critiques

**Comment by JonErvin** [comment]
Hey thanks @JTBassett !  Appreciate the patience on this one.  Caught one bug - the updated formulas in provider-level calculation only went down three rows (so when I switched the fourth row to Center, your updates didn't come through).  I went ahead and just dropped that formula all the way down the end of the sheet and it should be good.

Other than that I think we're set and ready to implement in production!

**Comment by JTBassett** [comment]
@csprayregen Jon and I moved this to the production spreadsheet - let us know if you have any questions!

**Comment by csprayregen** [comment]
Thanks to both of you! I followed only some of this, but it looks good to me, so I'll go ahead and close.
